lib/data/external/AwsClient.js

```diff
@@ -62,6 +62,7 @@ class AwsClient {
         const stream = request.createReadStream().on('error', err => {
             log.error('error streaming data from AWS', { error: err,
                 dataStoreName: this._dataStoreName });
+            return callback(err);
```

Comment:
In objectCopy, data.get is called directly, so the get error isn't caught and the API continues with the data.put, which makes the server hangs indefinitely